File: plugin/testData/shared/basics/Functions.shared.kt
package com.test

fun main(){
    println(addNumbers(left = 1, right = addNumbers2(2, addNumbers3(3, 4))))
}

fun addNumbers(left: Int, right: Int): Int = left + right
fun addNumbers2(left: Int, right: Int): Int {
    return left + right
}

fun addNumbers3(left: Int, right: Int): Int {
    fun subfunction(left: Int): Int {
        return left + right
    }
    return subfunction(left)
}

File: plugin/testDataOutputTs/shared/basics/Functions.shared.ts
//Package: com.test
//Converted using Khrysalis2


export function main(): void {
    console.log(addNumbers(1, addNumbers2(2, addNumbers3(3, 4))))
}
 
 

export function addNumbers(left: number, right: number): number {
    return left + right;
}
 

export function addNumbers2(left: number, right: number): number {
    return left + right
}
 
 

export function addNumbers3(left: number, right: number): number {
    subfunction(left: number): number {
        return left + right
    }
    ;
    return subfunction(left)
}
 
